cmd/lncli/cmd_open_channel.go

```diff
@@ -995,39 +993,61 @@ func sendFundingState(cancelCtx context.Context, cliCtx *cli.Context,
 }
 
 // finalizeMsgFromString creates the final message for the PsbtFinalize step
-// from either a hex encoded raw wire transaction or a base64 encoded PSBT
-// packet.
+// from either a hex encoded raw wire transaction or a base64/binary encoded
+// PSBT packet.
 func finalizeMsgFromString(tx string,
 	pendingChanID []byte) (*lnrpc.FundingTransitionMsg_PsbtFinalize, error) {
 
-	rawTx, err := hex.DecodeString(strings.TrimSpace(tx))
+	psbtBytes, err := decodePsbt(tx)
 	if err == nil {
-		// Hex decoding succeeded so we assume we have a raw wire format
-		// transaction. Let's submit that instead of a PSBT packet.
-		tx := &wire.MsgTx{}
-		err := tx.Deserialize(bytes.NewReader(rawTx))
-		if err != nil {
-			return nil, fmt.Errorf("deserializing as raw wire "+
-				"transaction failed: %v", err)
-		}
 		return &lnrpc.FundingTransitionMsg_PsbtFinalize{
 			PsbtFinalize: &lnrpc.FundingPsbtFinalize{
-				FinalRawTx:    rawTx,
+				SignedPsbt:    psbtBytes,
 				PendingChanId: pendingChanID,
 			},
 		}, nil
 	}
 
-	// If the string isn't a hex encoded transaction, we assume it must be
-	// a base64 encoded PSBT packet.
-	psbtBytes, err := base64.StdEncoding.DecodeString(strings.TrimSpace(tx))
+	// PSBT decode failed, try to parse it as a hex encoded Bitcoin
+	// transaction
+	rawTx, err := hex.DecodeString(strings.TrimSpace(tx))
+	if err != nil {
+		return nil, fmt.Errorf("hex decode failed: %w", err)
+	}
+	msgtx := &wire.MsgTx{}
+	err = msgtx.Deserialize(bytes.NewReader(rawTx))
 	if err != nil {
-		return nil, fmt.Errorf("base64 decode failed: %v", err)
+		return nil, fmt.Errorf("deserializing as raw wire "+
+			"transaction failed: %v", err)
 	}
 	return &lnrpc.FundingTransitionMsg_PsbtFinalize{
 		PsbtFinalize: &lnrpc.FundingPsbtFinalize{
-			SignedPsbt:    psbtBytes,
+			FinalRawTx:    rawTx,
 			PendingChanId: pendingChanID,
 		},
 	}, nil
 }
+
+// decodePsbt tries to decode the input as a binary or base64 PSBT. If this
+// succeeded, the PSBT bytes are returned, an error otherwise.
+func decodePsbt(psbt string) ([]byte, error) {
+	if strings.HasPrefix(psbt, "psbt\xff") {
+		// A binary PSBT (read from a file) always starts with the PSBT
+		// magic "psbt\xff" according to BIP 174
+		return []byte(psbt), nil
+	} else if strings.HasPrefix(strings.TrimSpace(psbt), "cHNidP") {
```

Comment:
nit: This would be a nice candidate for a `switch` statement with a `default` case. A bit more readable than if/else constructs IMO.

Reply:
the first if statement checks if the _raw_ input that could come from a file (`psbt`) is a binary psbt, the second one checks if whatever the user pasted (`strings.TrimSpace(psbt)`) is a base64 psbt. i see them as two different checks that happen to both look at the first few character. a new encoding that could be used in the future could be checked differently, and would not fit in a switch/case statement

Reply:
Yeah, I see that. But Golang allows you to just have boolean statements in `case` statements.
So this can become:

```
	switch {
	case strings.HasPrefix(psbt, "psbt\xff"):
		// A binary PSBT (read from a file) always starts with the PSBT
		// magic "psbt\xff" according to BIP 174
		return []byte(psbt), nil

	case strings.HasPrefix(strings.TrimSpace(psbt), "cHNidP"):
		// A base64 PSBT always starts with "cHNidP". This is the
		// longest base64 representation of the PSBT magic that is not
		// dependent on the byte after it.
		psbtBytes, err := base64.StdEncoding.DecodeString(
			strings.TrimSpace(psbt),
		)
		if err != nil {
			return nil, fmt.Errorf("base64 decode failed: %w", err)
		}

		return psbtBytes, nil

	default:
		return nil, fmt.Errorf("not a PSBT")
	}
```

Reply:
oh that looks clean, i had the javascript switch/case in mind